remindme-common/src/main/java/it/andreafailli/remindme/common/models/Reminder.java

```diff
@@ -50,5 +50,45 @@ public boolean isArchived() {
 	public void setArchived(boolean archived) {
 		this.archived = archived;
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + (archived ? 1231 : 1237);
+		result = prime * result + ((date == null) ? 0 : date.hashCode());
+		result = prime * result + ((title == null) ? 0 : title.hashCode());
+		result = prime * result + ((user == null) ? 0 : user.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)